client/src/modules/stock/exit/exit.js

```diff
@@ -228,6 +228,7 @@ function StockExitController(
           const depotExitType = ExitTypes.exitTypes.find(item => item.label === 'depot');
           onSelectExitType(depotExitType, destDepot);
           vm.destLabel = depotExitType.formatLabel(destDepot);
+          vm.validate();
```

Comment:
It seems like it does not work, what about wrapping the code inside `$timeout`, such as : 
```js
$timeout(() => {
  destDepot.shipment = vm.shipment;
  const depotExitType = ExitTypes.exitTypes.find(item => item.label === 'depot');
  onSelectExitType(depotExitType, destDepot);
  vm.destLabel = depotExitType.formatLabel(destDepot);
  vm.validate();
}, 0);
```

Reply:
I don't think the timeout is the best fix.    I suggest taking another look at the way the promises are resolved and how the loading flag is being handled.  If the loading is handled incorrectly, we could try loading the items in the list before all the items have arrived.   That sounds like what is going on...   I may have time to look at this again this evening, but not tomorrow.